core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -92,25 +98,25 @@ DeepCopy<Kokkos::Experimental::HIPSpace, HostSpace,
 
 DeepCopy<Kokkos::Experimental::HIPSpace, Kokkos::Experimental::HIPSpace,
          Kokkos::Experimental::HIP>::DeepCopy(const Kokkos::Experimental::HIP&
-                                              /*instance*/,
+                                                  instance,
                                               void* dst, const void* src,
                                               size_t n) {
-  // FIXME_HIP use instance
-  HIP_SAFE_CALL(hipMemcpy(dst, src, n, hipMemcpyDefault));
+  instance.fence();
+  DeepCopyAsyncHIP(dst, src, n);
```

Comment:
You've got to use the stream provided by the instance, not the one from `get_deep_copy_stream()`

Reply:
That's not how it works in CUDA. I can change it but then the behavior will be different between CUDA and HIP. No matter what's your instance, we always use the same stream for the `deep_copy`.

Reply:
Are you sure?
https://github.com/kokkos/kokkos/blob/b973b126183f6ea756c7b02591cecaa445e85a8f/core/src/Cuda/Kokkos_CudaSpace.cpp#L99-L103

What am I missing?

Reply:
Hmm I see. I used https://github.com/kokkos/kokkos/blob/develop/core/src/Kokkos_CudaSpace.hpp#L547-L551